### PR TITLE
ABC for `BooleanPolynomialRing`

### PR DESCRIPTION
**Pull request by mkoeppe**
### 📚 Description
The new ABC `sage.rings.polynomial.multi_polynomial_ring_base.BooleanPolynomialRing_base` can be imported for `isinstance` tests, without a compile time dependency on `brial`.

This is a step towards a modularized distribution **sagemath-brial**.

Part of: 
- #32414

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [x] I have linked an issue or discussion.
- [x] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/polynomial/pbori/pbori.pxd [hunk 1]
```diff
@@ -1,14 +1,13 @@
 from libcpp.memory cimport unique_ptr, shared_ptr, make_shared
 
-from sage.rings.polynomial.multi_polynomial_ring_base cimport \
-                                                MPolynomialRing_base
+from sage.rings.polynomial.multi_polynomial_ring_base cimport MPolynomialRing_base, BooleanPolynomialRing_base
```
Comment (kiwifb) [inline src/sage/rings/polynomial/pbori/pbori.pxd:3]: Is it still necessary to cimport  `MPolynomialRing_base`? It is the only instance in this file. Do any file importing from `sage.rings.polynomial.pbori.pbori` expect it? If so, could they import it directly from `sage.rings.polynomial.multi_polynomial_ring_base` instead?

Comment (mkoeppe) [reply src/sage/rings/polynomial/pbori/pbori.pxd:3]: It is used in `pbori.pyx` for an `isinstance` check

Comment (mkoeppe) [reply src/sage/rings/polynomial/pbori/pbori.pxd:3]: I can of course move the `cimport` to the `pyx` file if you think that's better

Comment (kiwifb) [reply src/sage/rings/polynomial/pbori/pbori.pxd:3]: I do not think it is particularly better. A cython expert would have to chime in on that. So, let's leave it as is.

**Comment by kiwifb** [comment]
I'll let the automated stuff do its bits before approving.

**Comment by mkoeppe** [comment]
Tests pass

**Review by kiwifb — APPROVED** [review]
I cannot see a reason not to approve.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**